docs-chef-io/content/inspec/resources/os.md

```diff
@@ -25,132 +25,207 @@ This resource first became available in v1.0.0 of InSpec.
+  it { should cmp.loose '22' }     # Passes for 22.04, 22.10, etc
+  it { should cmp.loose >= '20' }  # Passes for any version 20 or higher
+end
 
-Use the following helpers to test for operating system names, releases, and architectures:
+# Using segment matching for specific version parts
+describe os.version do
+  it { should cmp.segments(1) '22' }     # Only checks major version (22)
+  it { should cmp.segments(2) '22.04' }  # Checks major and minor version (22.04)
+  it { should cmp.segments(3) '22.04.1'} # Checks major, minor and patch version
+end
 
-    describe os.name do
-       it { should eq 'foo' }
+# Complex version comparisons
+control 'os-version' do
+  impact 1.0
+  title 'Verify OS version requirements'
+  desc 'Ensure OS version meets minimum requirements'
+  
+  if os.debian?
+    describe os.version do
+      it { should cmp >= '22.04' }
+      it { should cmp.segments(2) >= '22.04' }  # Must be Ubuntu 22.04 or higher
     end
-
-    describe os.release do
-       it { should eq 'foo' }
+  elsif os.redhat?
+    describe os.version do
+      it { should cmp >= '8.0' }
+      it { should cmp.loose >= '8' }  # Any RHEL 8 or higher
     end
+  end
+end
+```
 
-    describe os.arch do
-       it { should eq 'foo' }
-    end
+### Test Platform-Specific Information
 
-### os.family names
+```ruby
+# Test for Windows
+describe os do
+  it { should be_windows }
+  its('family') { should eq 'windows' }
+end
 
-Use `os.family` to enable more granular testing of platforms, platform names, architectures, and releases. Use any of the following platform-specific names to test for specific platforms:
+# Test for Ubuntu
+describe os do
+  it { should be_debian }
+  its('name') { should eq 'ubuntu' }
+  its('release') { should eq '22.04' }
+end
+```
 
-- `aix`
-- `bsd` For platforms that are part of the Berkeley OS family `darwin`, `freebsd`, `netbsd`, and `openbsd`.
-- `debian`
-- `hpux`
-- `linux`. For platforms that are part of the Linux family `alpine`, `amazon`, `arch`, `coreos`, `exherbo`, `fedora`, `gentoo`, and `slackware`.
-- `redhat`. For platforms that are part of the Redhat family `centos`.
-- `solaris`. For platforms that are part of the Solaris family `nexentacore`, `omnios`, `openindiana`, `opensolaris`, and `smartos`.
-- `suse`
-- `unix`
-- `windows`
+### Using the params Method
 
-For example, both of the following tests should have the same result:
+The `params` method returns a hash containing all platform information:
 
 ```ruby
-if os.family == 'debian'
-  describe port(69) do
-    its('processes') { should include 'in.tftpd' }
-  end
-elsif os.family == 'redhat'
-  describe port(69) do
-    its('processes') { should include 'xinetd' }
-  end
+describe os.params do
+  its(['name']) { should eq 'ubuntu' }
+  its(['family']) { should eq 'debian' }
+  its(['release']) { should eq '22.04' }
+  its(['arch']) { should eq 'x86_64' }
+  its(['version', 'major']) { should eq 22 }
+  its(['version', 'minor']) { should eq 4 }
```

Comment:
I recognize that most people wouldn't have this use case, but do you want to add 
`its(['version', 'patch']) { should eq 1 }`?